individual/services.py

```diff
@@ -607,41 +629,84 @@ def _validate_possible_individuals(self, dataframe: DataFrame, upload_id: uuid,
                 for field in unique_fields
             }
 
-        chunk_size = math.ceil(len(dataframe) / num_workers)
-        data_chunks = [dataframe[i:i + chunk_size] for i in range(0, dataframe.shape[0], chunk_size)]
+        check_location = 'location_name' in dataframe.columns
+        if check_location:
+            # Issue a single DB query instead of per row for efficiency
+            loc_name_dist_ids_from_db = self._query_location_district_ids(dataframe.location_name)
+            user_allowed_loc_ids = LocationManager().get_allowed_ids(self.user)
+            duplicate_village_names = self._query_dupcliate_village_names()
+        else:
+            loc_name_dist_ids_from_db = None
+            user_allowed_loc_ids = None
+            duplicate_village_names = None
 
-        validated_dataframe = []
-        with concurrent.futures.ProcessPoolExecutor(max_workers=num_workers) as executor:
-            futures = [executor.submit(
-                self.process_chunk, 
-                chunk, 
-                properties, 
-                unique_validations, 
-                calculation, 
-                calculation_uuid
-            ) for chunk in data_chunks]
-            
-            for future in concurrent.futures.as_completed(futures):
-                validated_dataframe.extend(future.result())
+        # TODO: Use ProcessPoolExecutor after resolving django dependency loading issue
+        validated_dataframe = IndividualImportService.process_chunk(
+            dataframe,
+            properties,
+            unique_validations,
+            loc_name_dist_ids_from_db,
+            user_allowed_loc_ids,
+            duplicate_village_names,
+        )
 
         self.save_validation_error_in_data_source_bulk(validated_dataframe)
         invalid_items = fetch_summary_of_broken_items(upload_id)
         return validated_dataframe, invalid_items
 
-    def _handle_uniqueness(self, row, field, field_properties, dataframe):
-        unique_class_validation = IndividualConfig.unique_class_validation
-        calculation_uuid = IndividualConfig.validation_calculation_uuid
-        calculation = get_calculation_object(calculation_uuid)
-        result_row = calculation.calculate_if_active_for_object(
-            unique_class_validation,
-            calculation_uuid,
-            field_name=field,
-            field_value=row[field],
-            incoming_data=dataframe
+    @staticmethod
+    def _query_location_district_ids(location_names):
+        locations = Location.objects.filter(type="V", name__in=location_names.unique().tolist())
+        return {loc.name: loc.parent.parent.id for loc in locations}
```

Comment:
this is about name not ID, also in LMIC it happen quite often that many villlages share the same names, I would rather have a resolver of name to id only if location is not a number

Reply:
I think code should be used instead, if you want to use names then ask for village, municipality, district, region names
as done in im_export module